### PR TITLE
Add DidYouMean for HasManyThroughAssociationNotFoundError

### PR DESCRIPTION
**Pull request by p8**
### Summary

If a has_many :through association isn't found we can suggest similar associations:

```
class Author
  has_many :categorizations, -> { }
  has_many :categories, through: :categorizations
  has_many :categorized_posts, through: :categorizations, source: :post
  has_many :category_post_comments, through: :categories, source: :post_comments

  has_many :nothings, through: :kateggorisatons, class_name: "Category"
end

Author.first.nothings

Could not find the association :kateggorisatons in model Author
Did you mean?  categorizations
               categories
               categorized_posts
               category_post_comments
```

### CODE REVIEW

File: activerecord/lib/active_record/associations.rb [hunk 1]
```diff
@@ -51,13 +51,41 @@ def initialize(reflection = nil, associated_class = nil)
...
       end
     end
+
+    class Correction
+      def initialize(error)
+        @error = error
+      end
+
+      def corrections
+
```
Comment (simi) [inline activerecord/lib/active_record/associations.rb:71]: :scissors:

File: activerecord/test/cases/associations/join_model_test.rb [hunk 2]
```diff
@@ -341,6 +341,16 @@ def test_unavailable_through_reflection
     assert_raise(ActiveRecord::HasManyThroughAssociationNotFoundError) { authors(:david).nothings }
   end
 
+  if defined?(DidYouMean) && DidYouMean.respond_to?(:correct_error)
+    def test_exceptions_have_suggestions_for_fix
+      error = assert_raise(ActiveRecord::HasManyThroughAssociationNotFoundError) {
+        authors(:david).nothings
+      }
+      puts error.message
```
Comment (tenderlove) [inline activerecord/test/cases/associations/join_model_test.rb:349]:
```suggestion
```

File: activerecord/lib/active_record/associations.rb (hunk 1 above)
Comment (p8) [reply activerecord/lib/active_record/associations.rb]: Fixed!

File: activerecord/test/cases/associations/join_model_test.rb (hunk 2 above)
Comment (p8) [reply activerecord/test/cases/associations/join_model_test.rb]: Fixed!

**Pull request merged**